conmon/conmon.c

```diff
@@ -56,29 +56,53 @@
 	} while (0)
 
-#define nwarn(s)                                             \
-	do {                                                 \
-		fprintf(stderr, "[conmon:w]: %s\n", s);      \
-		syslog(LOG_INFO, "conmon <nwarn>: %s\n", s); \
-	} while (0)
+#define nwarn(s)                                                     \
+	if (parse_level(opt_log_level) > 0) {                                     \
+		do {                                                 \
+			fprintf(stderr, "[conmon:w]: %s\n", s);      \
+			syslog(LOG_INFO, "conmon <nwarn>: %s\n", s); \
+		} while (0);                                         \
+	}
 
-#define nwarnf(fmt, ...)                                                       \
-	do {                                                                   \
-		fprintf(stderr, "[conmon:w]: " fmt "\n", ##__VA_ARGS__);       \
-		syslog(LOG_INFO, "conmon <nwarn>: " fmt " \n", ##__VA_ARGS__); \
-	} while (0)
+#define nwarnf(fmt, ...)                                                               \
+	if (parse_level(opt_log_level) > 0) {                                                       \
+		do {                                                                   \
+			fprintf(stderr, "[conmon:w]: " fmt "\n", ##__VA_ARGS__);       \
+			syslog(LOG_INFO, "conmon <nwarn>: " fmt " \n", ##__VA_ARGS__); \
+		} while (0);                                                           \
+	}
 
-#define ninfo(s)                                             \
-	do {                                                 \
-		fprintf(stderr, "[conmon:i]: %s\n", s);      \
-		syslog(LOG_INFO, "conmon <ninfo>: %s\n", s); \
-	} while (0)
+#define ninfo(s)                                                     \
+	if (parse_level(opt_log_level) > 1) {                                     \
+		do {                                                 \
+			fprintf(stderr, "[conmon:i]: %s\n", s);      \
+			syslog(LOG_INFO, "conmon <ninfo>: %s\n", s); \
+		} while (0);                                         \
+	}
 
-#define ninfof(fmt, ...)                                                       \
-	do {                                                                   \
-		fprintf(stderr, "[conmon:i]: " fmt "\n", ##__VA_ARGS__);       \
-		syslog(LOG_INFO, "conmon <ninfo>: " fmt " \n", ##__VA_ARGS__); \
-	} while (0)
+#define ninfof(fmt, ...)                                                               \
+	if (parse_level(opt_log_level) > 1) {                                                       \
+		do {                                                                   \
+			fprintf(stderr, "[conmon:i]: " fmt "\n", ##__VA_ARGS__);       \
+			syslog(LOG_INFO, "conmon <ninfo>: " fmt " \n", ##__VA_ARGS__); \
+		} while (0);                                                           \
+	}
+
+#define ndebug(s)                                                     \
+	if (parse_level(opt_log_level) > 2) {                                      \
```

Comment:
2 doesn't match the enum values you have defined below. We should use the enum constants instead of number in this comparison.

Reply:
Fixed.